src/nft/hooks/useBag.ts

```diff
@@ -52,19 +55,22 @@ export const useBag = create<BagState>()(
         set(() => ({
           didOpenUnavailableAssets: didOpen,
         })),
-      bagExpanded: false,
       toggleBag: () =>
-        set(({ bagExpanded }) => ({
-          bagExpanded: !bagExpanded,
-        })),
+        set(({ bagExpanded }) => {
+          if (bagExpanded === false) {
+            return { bagExpanded: !bagExpanded, bagManuallyClosed: true }
```

Comment:
I don't think this is valid logic - sometimes we close the bag on purpose for instance if a user goes to another page we might toggle the bag - also if bagExpanded is false that means we are setting the bag to open now. Maybe a better way to go about this is by passing an optional param for manualClose that sets manually close if its there